src/dali/abstract_ccxt_input_plugin.py

```diff
@@ -0,0 +1,651 @@
+
+    def _get_process_deposits_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        raise NotImplementedError("Abstract method")
+
+    def _get_process_withdrawals_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        raise NotImplementedError("Abstract method")
+
+    # Some exchanges require you to loop through all markets for trades
+    def _get_process_trades_pagination_detail_set(self) -> Optional[AbstractPaginationDetailSet]:
+        raise NotImplementedError("Abstract method")
+
+    def _get_markets(self) -> List[str]:
+
+        if self.__markets:
+            return self.__markets
+
+        ccxt_markets: Any = self._client.fetch_markets()
+        market_list: List[str] = []
+        for market in ccxt_markets:
+            self._logger.debug("Market: %s", json.dumps(market))
+            market_list.append(market[_ID])
+
+        self.__markets = market_list
+
+        return self.__markets
+
+    @staticmethod
+    def _rp2_timestamp_from_ms_epoch(epoch_timestamp: str) -> str:
+        rp2_time = datetime.fromtimestamp((int(epoch_timestamp) / _MS_IN_SECOND), timezone.utc)
+
+        return rp2_time.strftime("%Y-%m-%d %H:%M:%S%z")
+
+    # Parses the symbol (eg. 'BTC/USD') into base and quote assets, and formats notes for the transactions
+    @staticmethod
+    def _to_trade(market_pair: str, base_amount: str, quote_amount: str) -> Trade:
+        assets = market_pair.split("/")
+        return Trade(
+            base_asset=assets[0],
+            quote_asset=assets[1],
+            base_info=f"{base_amount} {assets[0]}",
+            quote_info=f"{quote_amount} {assets[1]}",
+        )
+
+    def load(self) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+        in_transactions: List[InTransaction] = []
+        out_transactions: List[OutTransaction] = []
+        intra_transactions: List[IntraTransaction] = []
+
+        if self._client.has[_FETCH_DEPOSITS]:
+            self._process_deposits(intra_transactions)
+        self._process_gains(in_transactions, out_transactions)
+        if self._client.has[_FETCH_MY_TRADES]:
+            self._process_trades(in_transactions, out_transactions)
+        if self._client.has[_FETCH_WITHDRAWALS]:
+            self._process_withdrawals(intra_transactions)
+        self._process_implicit_api(in_transactions, out_transactions, intra_transactions)
+
+        result.extend(in_transactions)
+        result.extend(out_transactions)
+        result.extend(intra_transactions)
+
+        return result
+
+    ### Multiple Transaction Processing or Pagination Methods
+
+    def _process_deposits(
+        self,
+        intra_transactions: List[IntraTransaction],
+    ) -> None:
+
+        pagination_detail_set: Optional[AbstractPaginationDetailSet] = self._get_process_deposits_pagination_detail_set()
+        # Strip optionality
+        if not pagination_detail_set:
```

Comment:
Shouldn't we throw an exception here, instead of logging? If `pagination_detail_set` is `None` calling `iter` on it a few lines down will cause a crash.